$MirOS: ports/net/curl/patches/patch-docs_examples_ftpupload_c,v 1.1 2008/04/05 21:54:38 tg Exp $
--- docs/examples/ftpupload.c.orig	Fri May 23 09:05:31 2008
+++ docs/examples/ftpupload.c	Tue Aug 26 19:03:21 2008
@@ -21,6 +21,7 @@
 #else
 #include <unistd.h>
 #endif
+#include <string.h>
 
 /*
  * This example shows an FTP upload, with a rename of the file just after
@@ -45,7 +46,7 @@ static size_t read_callback(void *ptr, s
      by default internally */
   size_t retcode = fread(ptr, size, nmemb, stream);
 
-  fprintf(stderr, "*** We read %d bytes from file\n", retcode);
+  fprintf(stderr, "*** We read %zd bytes from file\n", retcode);
   return retcode;
 }
 
@@ -65,7 +66,7 @@ int main(int argc, char **argv)
     printf("Couldnt open '%s': %s\n", LOCAL_FILE, strerror(errno));
     return 1;
   }
-  printf("Local file size: %ld bytes.\n", file_info.st_size);
+  printf("Local file size: %lld bytes.\n", (long long)file_info.st_size);
 
   /* get a FILE * of the same file */
   hd_src = fopen(LOCAL_FILE, "rb");
